lib/openhab/dsl/items/item_command.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+require 'java'
+
+module OpenHAB
+  module DSL
+    module Items
+      #
+      # Holds methods to automatically generate commands and
+      # accessors for items
+      module ItemCommand
+        #
+        # For every value in the supplied enumeration create a corresponding method mapped to the lowercase
+        # string representation of the enum value For example, an enum with values of STOP and START
+        # would create methods stop() and start() that send the corresponding STOP and START commands to the item
+        #
+        # @param [Java::JavaLang::Enum] command_enum Enumeration to create commands for
+        #
+        #
+        def item_command(command_enum)
+          # rubocop:disable Style/HashEachMethods
+          # Disable rule because Java enum does not support each_value
+          command_enum.values.each do |command|
+            command_method = command.to_s.downcase
+            define_method(command_method) do
+              self.command(command)
+            end
+          end
+          # rubocop:enable Style/HashEachMethods
+        end
+
+        #
+        # For every value in the supplied enumeration create a corresponding method mapped to the lowercase
+        # string representation appended with a question mark '?' of the enum value For example,
+        # an enum with values of UP and DOWN would create methods up? and down? that check
+        # if the current state matches the value of the enum
+        #
+        # @param [Java::JavaLang::Enum] command_enum Enumeration to create methods for each value
+        #   to check if current state matches that enum
+        # @yield [state] Optional block that can be used to transform state prior to comparison
+        #
+        #
+        def item_state(command_enum)
+          # rubocop:disable Style/HashEachMethods
+          # Disable rule because Java enum does not support each_value
+          command_enum.values.each do |command|
+            status_method = "#{command.to_s.downcase}?"
+            define_method(status_method) do
+              if state?
+                current_state = state
```

Comment:
Would it be possible to use `current_state = state.as(command_enum)`? Then we could have a method where you can pass all supported types as an array.

Reply:
Would that always work for all cases to do it as default? Right now I have the ability to pass in a block to do that, but if that isn't needed then it would be better.  I didn't realize we wanted to do that on all types since the others (besides rollershutter) seemed to just work.

Reply:
Just thinking that if we should make things as generic as possible it would be easier. Or are there any possible cases where the block would contain anything else? And even if the state already is a e.g PercentType, it won't cause any problems.

Reply:
The other item types we have only supports one type of state (except UndDefType), so they wouldn't have the same problem. But should be useable to add OnOffType to DimmerItems for example

Reply:
Looks like it does work everywhere..  code updated.